gzip_logs_tools/log_compressor/log_compressor.c

```diff
@@ -810,6 +812,11 @@ static thread_func_t threads[] = {
 static bool_t
 file_mode_main(const char* path)
 {
+	if (strlen(path) > sizeof(struct sockaddr_un) - 1)
+	{
+		log_print("init_unix_dgram_socket: path %s too long", path);
+		return FALSE;
+	}
```

Comment:
from what i see, if we put this after the unlink, we don't have to disable the warning on strncpy